File: Semester 3/MAP/Seminar10/Seminar10/container/AbstractContainer.cs
using Task = Seminar10.domain.Task;

namespace Seminar10.container;

public abstract class AbstractContainer: Container
{
    protected List<Task> tasks;
    
    public AbstractContainer ()
    {
        tasks = new List<Task>();
    }

    public virtual Task remove()
    {
        throw new NotImplementedException();
    }
    
    public void add(Task task){
        tasks.Add(task);
    }

    public bool isEmpty()
    {
        return tasks.Count == 0;
    }

    public int size()
    {
        return tasks.Count;
    }
}

File: Semester 3/MAP/Sem1_pb1_MAP/src/factory/AbstractContainer.java
package factory;

import model.Task;

import java.util.ArrayList;

public interface AbstractContainer extends Container {
    ArrayList<Task> list=new ArrayList<>();

    @Override
    public abstract Task remove() ;

    @Override
    public default void add(Task task) {
        list.add(task);
    }

    @Override
    public default int size() {
        return list.size();
    }

    @Override
    public default boolean isEmpty() {
        return list.isEmpty();
    }
}
